DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -1615,7 +1615,13 @@ public DMASTProcStatementLabel Label(DMASTIdentifier expression)
                         } else {
                             Error("Expected ':'");
                         }
-                    } else {
+                    } else if (b is DMASTListIndex idx && idx.Expression is DMASTDereference idxDeref && idxDeref.Expression is DMASTListIndex)
```

Comment:
This doesn't work with expressions like `prob(50)?L[1]:L[2]:L[3]`. It should be parsed as `prob(50) ? L[1] : L[2]:L[3]` but is instead parsed as `prob(50) ? L[1]:L[2] : L[3]`.

This is the reason the other case above uses a do/while loop. It'd be nice if this could somehow be handled within that same loop, but I'm not sure what the best way to do that would be.